packages/compiler/core/checker.ts

```diff
@@ -2097,34 +2067,31 @@ export function createChecker(program: Program): Checker {
     }
   }
 
-  function getMergedSymbol<T extends Sym>(sym: T | undefined): T | undefined {
-    if (!sym) return sym;
+  function getMergedSymbol<T extends Sym>(sym: T | undefined): T {
```

Comment:
I thought about, but I think I should probably just remove "undefined" from the parameter type since it's less useful now that `symbol` is non-optional.